client/components/app.jsx

```diff
@@ -1,21 +1,52 @@
 import React from 'react';
 import Header from './header';
 import ProductList from './product-list';
+import ProductDetails from './product-details';
 export default class App extends React.Component {
   constructor(props) {
     super(props);
+    this.state = {
+      view: {
+        name: 'catalog',
+        params: {}
+      }
+    };
+    this.setView = this.setView.bind(this);
+  }
 
+  setView(name, params) {
+    this.setState({
+      view: {
+        name,
+        params
+      }
+    });
   }
 
-  render() {
-    return (
-      <div>
-        <Header/>
-        <div className="container">
-          <ProductList/>
+  renderViewer() {
+    if (this.state.view.name === 'catalog') {
+      return (
+        <div>
+          <Header/>
+          <div className="container">
+            <ProductList setView={this.setView}/>
+          </div>
+        </div>
+
+      );
+    } else {
+      return (
+        <div>
+          <Header/>
+          <div className="container">
+            <ProductDetails productInfo={this.state.view.params} setView={this.setView}/>
+          </div>
         </div>
```

Comment:
This method should only return the parts that are different. The parts that are the same can be left in `render()`